test/cpp/test_xla_sharding.cpp

```diff
@@ -64,5 +64,81 @@ TEST_F(XLAShardingTest, ShardTensor) {
   EXPECT_EQ(shards[7].sizes(), c10::ArrayRef<long>({8, 7, 4}));
 }
 
+TEST_F(XLAShardingTest, CreateTensorsData) {
+  if (xla::sys_util::GetEnvString(xla::env::kEnvPjRtDevice, "") == "") {
+    GTEST_SKIP() << "`PJRT_DEVICE` is not set.";
+  }
+
+  std::vector<at::Tensor> tensors(2);
+  std::fill_n(tensors.begin(), tensors.size(),
+              at::ones({8, 8}, at::TensorOptions(at::kFloat)));
+  std::vector<std::string> devices(2);
+  std::fill_n(devices.begin(), devices.size(), GetDefaultDevice()->toString());
+  std::vector<XLATensor::ShardingSpecPtr> shardings = {
```

Comment:
I really think we should break these into 2 tests.

Reply:
I see, could you elaborate more? If it's the FUNCTION_IN_TEST related and a couple of cases (sharding & replicated), then we can keep it as one. I think optionally we can break this into two, per a style preference. But I could be missing something.

Reply:
I think we should have one `TestNullSharding` and one `TestReplicated`. Each test should only test one behavior.